logstash-core/lib/logstash/plugin.rb

```diff
@@ -31,9 +49,21 @@ class LogStash::Plugin
   # }
   # ```
   #
-  # If you don't explicitely set this variable Logstash will generate a unique name.
+  # If you don't explicitly set this variable Logstash will generate a unique name.
   config :id, :validate => :string
 
+  # Add a set of specific operation timing thresholds for the plugin, they are used in the context
+  # of reporting slow operations.
+  #
+  # ```
+  #  thresholds => {
+  #    filter.grok.runtime => 12
+  #  }
+  # ```
+  #
+  # The values defined here takes precedence over the ones defined in the logstash config file. (default => {})
+  config :thresholds, :validate => :hash, :default => {}
```

Comment:
I don't know if all plugins will have "thresholds" -- and I want to learn from my mistake of making `codec` part of the base input and output classes (not all of those plugins can even use codecs).  Do all plugins need a thresholds config setting? What if a plugin doesn't implement any thresholds at all, I feel we shouldn't show this setting to the user if a plugin doesn't support it.